src/main/kotlin/com/team4099/robot2023/subsystems/elevator/ElevatorIOKraken.kt

```diff
@@ -0,0 +1,118 @@
+        elevatorLeaderKraken.configurator.apply(elevatorLeaderConfiguration)
+        elevatorFollowerKraken.configurator.apply(elevatorFollowerConfiguration)
+        elevatorLeaderKraken.inverted = true
+        elevatorFollowerKraken.inverted = false
+        MotorChecker.add("Elevator", "Extension", MotorCollection(mutableListOf(
+                Falcon500(elevatorLeaderKraken, "Leader Extension Motor"), (Falcon500(elevatorFollowerKraken, "Follower Extension Motor"))), ElevatorConstants.LEADER_STATOR_CURRENT_LIMIT, 30.celsius, ElevatorConstants.LEADER_STATOR_CURRENT_LIMIT-30.amps, 110.celsius))
+        elevatorLeaderStatorCurrentSignal = elevatorLeaderKraken.statorCurrent
+        elevatorLeaderSupplyCurrentSignal = elevatorLeaderKraken.supplyCurrent
+        elevatorLeadertempSignal = elevatorLeaderKraken.deviceTemp
+        elevatorLeaderDutyCycle = elevatorLeaderKraken.dutyCycle
+        elevatorFollowerStatorCurrentSignal = elevatorFollowerKraken.statorCurrent
+        elevatorFollowerSupplyCurrentSignal = elevatorFollowerKraken.supplyCurrent
+        elevatorFollowertempSignal = elevatorFollowerKraken.deviceTemp
+        elevatorFollowerDutyCycle = elevatorFollowerKraken.dutyCycle
+    }
+
+    override fun configPID(kP: ProportionalGain<Meter, Volt>, kI: IntegralGain<Meter, Volt>, kD: DerivativeGain<Meter, Volt>) {
+        val pidController = Slot0Configs()
+        pidController.kP = leaderSensor.proportionalPositionGainToRawUnits(kP)
+        pidController.kI = leaderSensor.integralPositionGainToRawUnits(kI)
+        pidController.kD = leaderSensor.derivativePositionGainToRawUnits(kD)
+    }
+
+    override fun updateInputs(inputs: ElevatorIO.ElevatorInputs) {
+        inputs.elevatorPosition = leaderSensor.position
+        inputs.elevatorVelocity = leaderSensor.velocity
+        inputs.leaderAppliedVoltage = elevatorLeaderDutyCycle.value.volts
+        inputs.followerAppliedVoltage = elevatorFollowerDutyCycle.value.volts
+        inputs.leaderSupplyCurrent = elevatorLeaderSupplyCurrentSignal.value.amps
+        inputs.leaderStatorCurrent = elevatorLeaderStatorCurrentSignal.value.amps
+        inputs.followerSupplyCurrent = elevatorFollowerSupplyCurrentSignal.value.amps
+        inputs.followerStatorCurrent = elevatorFollowerStatorCurrentSignal.value.amps
+        inputs.leaderTempCelcius = elevatorLeadertempSignal.value.celsius
+        inputs.followerTempCelcius = elevatorFollowertempSignal.value.celsius
+        inputs.leaderRawPosition = leaderSensor.getRawPosition()
+        inputs.followerRawPosition = followerSensor.getRawPosition()
+    }
+
+    override fun setOutputVoltage(voltage: ElectricalPotential) {
+        elevatorLeaderKraken.setVoltage(voltage.inVolts)
+    }
+
+    override fun setPosition(position: Length, feedForward: ElectricalPotential) {
+        elevatorLeaderKraken.setControl(PositionDutyCycle(leaderSensor.positionToRawUnits(position), 0.0, true, feedForward.inVolts, 0, false, false, false))
```

Comment:
you want to use `PositionVoltage` here not `PositionDutyCycle`. The difference being `PositionDutyCycle` accepting a feedforward in terms of power units [-1,1] and `PositionVoltage` accepting a feedforward in terms of volts (which is what we're using)

Reply:
also use the wrapper class we have

Reply:
![image](https://github.com/team4099/Crescendo-2024/assets/67883747/504d8d84-ce97-4214-9c44-e8cea1bab5b2)
how do we change this from length to angle?